src/sqlfluff/config.py

```diff
@@ -265,9 +265,29 @@ def load_config_up_to_path(self, path):
         """Loads a selection of config files from both the path and it's parent paths."""
         user_appdir_config = self.load_user_appdir_config()
         user_config = self.load_user_config()
+        config_paths = self.find_config_up_to_path(path)
+        config_stack = [self.load_config_at_path(p) for p in config_paths]
+        return nested_combine(user_appdir_config, user_config, *config_stack)
+
+    def find_ignore_config_files(
+        self, path, working_path=os.getcwd(), ignore_file_name=".sqlfluffignore"
+    ):
+        """Finds sqlfluff ignore files from both the path and its parent paths."""
+        return set(
+            filter(
+                os.path.isfile,
+                map(
+                    lambda x: os.path.join(x, ignore_file_name),
+                    self.find_config_up_to_path(path=path, working_path=working_path),
+                ),
+            )
+        )

```

Comment:
I like this pattern, but I still feel a bit weird about the isolation, and having it instantiate the loader. I think I'm just concerned about what happens when people want to start passing configs in from other places.

I think what would make me feel more comfortable, would be if this was instead a `classmethod`, the only thing it depends on is `find_config_up_to_path`, and I think that method based on your existing changes should actually be a `staticmethod` (on top of my other comments below).

That way when we actually want to use this method in `linter.py` we never have to instantiate it, and we can just call `ConfigLoader.find_ignore_config_files()`.